pkg/crc/errors/multierror.go

```diff
@@ -22,12 +22,30 @@ func (m MultiError) ToError() error {
 	if len(m.Errors) == 0 {
 		return nil
 	}
+	if len(m.Errors) == 1 {
+		return m.Errors[0]
+	}
+
+	var aggregatedErrors []string
 
-	errStrings := []string{}
-	for _, err := range m.Errors {
-		errStrings = append(errStrings, err.Error())
+	count := 1
+	current := m.Errors[0].Error()
+	for i := 1; i < len(m.Errors); i++ {
+		if m.Errors[i].Error() == current {
```

Comment:
Maybe also check `reflect.TypeOf` in addition to the string value?

Reply:
Error() always return a string. I don't see what you mean.

Reply:
I was wondering if something like this on top of your changes could be useful so that we check both error type and string:

```
diff --git a/pkg/crc/errors/multierror.go b/pkg/crc/errors/multierror.go
index a7d11afd..a28f5866 100644
--- a/pkg/crc/errors/multierror.go
+++ b/pkg/crc/errors/multierror.go
@@ -2,6 +2,7 @@ package errors
 
 import (
 	"fmt"
+	"reflect"
 	"strings"
 	"time"
 
@@ -12,6 +13,25 @@ type MultiError struct {
 	Errors []error
 }
 
+type errorWithCount struct {
+	errorCount int
+	err        error
+}
+
+func (err *errorWithCount) Error() string {
+	if err.errorCount == 1 {
+		return err.err.Error()
+	}
+	return fmt.Sprintf("%s (x%d)", err.err.Error(), err.errorCount)
+}
+
+func equalErr(err1, err2 error) bool {
+	if reflect.TypeOf(err1) != reflect.TypeOf(err2) {
+		return false
+	}
+	return err1.Error() == err2.Error()
+}
+
 func (m MultiError) Error() string {
 	if len(m.Errors) == 0 {
 		return ""
@@ -20,22 +40,27 @@ func (m MultiError) Error() string {
 		return m.Errors[0].Error()
 	}
 
-	var aggregatedErrors []string
+	var aggregatedErrors []error
 
 	count := 1
-	current := m.Errors[0].Error()
+	current := m.Errors[0]
 	for i := 1; i < len(m.Errors); i++ {
-		if m.Errors[i].Error() == current {
+		if equalErr(m.Errors[i], current) {
 			count++
 			continue
 		}
-		aggregatedErrors = append(aggregatedErrors, errorWithCount(current, count))
+		aggregatedErrors = append(aggregatedErrors, &errorWithCount{errorCount: count, err: current})
 		count = 1
-		current = m.Errors[i].Error()
+		current = m.Errors[i]
 	}
-	aggregatedErrors = append(aggregatedErrors, errorWithCount(current, count))
+	aggregatedErrors = append(aggregatedErrors, &errorWithCount{errorCount: count, err: current})
 
-	return strings.Join(aggregatedErrors, "\n")
+	var strBuilder strings.Builder
+	for _, err := range aggregatedErrors {
+		strBuilder.WriteString(err.Error())
+		strBuilder.WriteString("\n")
+	}
+	return strBuilder.String()
 }
 
 func (m *MultiError) Collect(err error) {
@@ -44,13 +69,6 @@ func (m *MultiError) Collect(err error) {
 	}
 }
 
-func errorWithCount(current string, count int) string {
-	if count == 1 {
-		return current
-	}
-	return fmt.Sprintf("%s (x%d)", current, count)
-}
-
 // RetriableError is an error that can be tried again
 type RetriableError struct {
 	Err error
```
(this does not pass `make test` because of an extra `\n` at the end of `Error()` and I'd actually move most of this to `Collect()` rather than `Error()` - maybe I'll do all of that as a followup)